src/auto/uri.rs

```diff
@@ -0,0 +1,530 @@
+        unsafe { from_glib_none(glib_sys::g_uri_get_auth_params(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn get_flags(&self) -> UriFlags {
+        unsafe { from_glib(glib_sys::g_uri_get_flags(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn get_fragment(&self) -> Option<GString> {
+        unsafe { from_glib_none(glib_sys::g_uri_get_fragment(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn get_host(&self) -> Option<GString> {
+        unsafe { from_glib_none(glib_sys::g_uri_get_host(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn get_password(&self) -> Option<GString> {
+        unsafe { from_glib_none(glib_sys::g_uri_get_password(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn get_path(&self) -> Option<GString> {
+        unsafe { from_glib_none(glib_sys::g_uri_get_path(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn get_port(&self) -> i32 {
+        unsafe { glib_sys::g_uri_get_port(self.to_glib_none().0) }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn get_query(&self) -> Option<GString> {
+        unsafe { from_glib_none(glib_sys::g_uri_get_query(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn get_scheme(&self) -> Option<GString> {
+        unsafe { from_glib_none(glib_sys::g_uri_get_scheme(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn get_user(&self) -> Option<GString> {
+        unsafe { from_glib_none(glib_sys::g_uri_get_user(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn get_userinfo(&self) -> Option<GString> {
+        unsafe { from_glib_none(glib_sys::g_uri_get_userinfo(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn parse_relative(&self, uri_ref: &str, flags: UriFlags) -> Result<Uri, Error> {
+        unsafe {
+            let mut error = ptr::null_mut();
+            let ret = glib_sys::g_uri_parse_relative(
+                self.to_glib_none().0,
+                uri_ref.to_glib_none().0,
+                flags.to_glib(),
+                &mut error,
+            );
+            if error.is_null() {
+                Ok(from_glib_full(ret))
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    fn to_string(&self) -> GString {
+        unsafe { from_glib_full(glib_sys::g_uri_to_string(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(feature = "v2_66", feature = "dox"))]
+    pub fn to_string_partial(&self, flags: UriHideFlags) -> Option<GString> {
```

Comment:
@elmarco Can this return `NULL`? Probably not? Should get a `(not nullable)` annotation in GLib then and the corresponding configuration in `Gir.toml`

Reply:
See https://gitlab.gnome.org/GNOME/glib/-/merge_requests/1687

Reply:
So this method shouldn't return an Option<T>?

Reply:
Yes, `nullable = false` for the return value in `Gir.toml` is needed here. Also see the various other functions marked as not nullable in that MR.